python/lsst/analysis/tools/actions/vector/vectorActions.py

```diff
@@ -52,6 +53,33 @@ def __call__(self, data: KeyedData, **kwargs) -> Vector:
         return cast(Vector, data[self.vectorKey.format(**kwargs)])[mask]
 
 
+class MultiCriteriaDownselectVector(VectorAction):
+    """Get a vector from KeyedData, apply specified set of selectors with AND logic, and return the
+    shorter Vector.
+    """
+
+    vectorKey = Field[str](doc="column key to load from KeyedData")
+
+    selectors = ConfigurableActionStructField[VectorAction](
+        doc="Selectors for selecting rows, will be AND together",
+    )
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        yield (self.vectorKey, Vector)
+        for action in self.selectors:
+            yield from cast(VectorAction, action).getInputSchema()
+
+    def __call__(self, data: KeyedData, **kwargs) -> Vector:
+        mask: Optional[Vector] = None
+        for selector in self.selectors:
+            subMask = selector(data, **kwargs)
+            if mask is None:
+                mask = subMask
+            else:
+                mask *= subMask  # type: ignore
+        return cast(Vector, data[self.vectorKey.format(**kwargs)])[mask]
+
+
```

Comment:
The `DownselectVector` returns a vector where a single configurable action has been used to apply selection criteria. Usually this will be a shorter vector due to the applied filtering.

In this instance, I needed to apply several per-group selection criteria (e.g., number of repeated measurements, mean signal-to-noise ratio) to keep only the per-group signal-to-noise measurements of interest. 

The first time I implemented this, I introduced the `AndSelector`, but then I felt that this made the syntax more complicated/verbose than I wanted. Basically I needed to introduce the new AndSelector and then set the multiple selectors for that selector. It added an extra layer of nesting that seemed unwarranted. 

I also considered to make `DownselectVector` always allow for the possibility of multiple selectors by using `selectors = ConfigurableActionStructField` rather than `selector = ConfigurableActionField`. This also seemed unsatisfying because one would need to add an extra layer in cases where there is only a single criterion applied. For example

```
self.process.filterActions.xStars = DownselectVector(
    vectorKey="mags", selector=VectorSelector(vectorKey="starSelector")
)
```

would become

```
self.process.filterActions.xStars = DownselectVector(vectorKey="mags")
self.process.filterActions.xStars.selectors.star = VectorSelector(vectorKey="starSelector")
```

I wasn't sure that we wanted to introduce that additional layer.

I'm still not fully convinced that introducing `MultiCriteriaDownselectVector` is the optimal solution.

Reply:
I might be remembering the And selector you had, but I could have sworn there was something different elsewhere. Anyway, the way I had imagined this to work is that you would build the mask you want (with all the various selectors) in the build stage, and then you would just apply that mask in the filter stage, and thus you would only need the one selector (`VectorSelector`), and then you would have the vector around for anything else you might need it for. I am not saying things MUST be done how I thought them out, just that that process made sense to me reading things out. I wanted to bring that up to the discussion for consideration, but if you determine your way is best I am happy to have the new selector, but it does make the other kind of redundant.